### PR TITLE
Add shell completion

### PR DESCRIPTION
**Pull request by ozzieba**
Follows https://github.com/spf13/cobra/blob/master/bash_completions.md
to resolve Issue #299

### CODE REVIEW

**Comment by rndstr** [comment]
nice, thanks!

would you be able to also add support for zsh? that would be `GenZshCompletion()`. maybe we can have `eksctl completion <bash|zsh>`.

would it make sense to add a hint about this to the README?

**Comment by ozzieba** [comment]
I'm not sure how to put completion in its own package, since that would create a cyclic dependency between `main` and `completion` (since I need `rootCmd`)...

**Comment by ozzieba** [comment]
Hmm, I suppose I could just export a function that takes an arbitrary Command and returns a `completion` command for it

**Comment by ozzieba** [comment]
Not really sure the refactored version is better, let me know what you think...

File: README.md [hunk 1]
```diff
@@ -331,6 +331,19 @@ eksctl apply --cluster-config advanced-cluster.yaml
...
+### Shell Completion
+
+To enable bash completion, run the following, or put it in `~/.bashrc` or `~/.profile`:
+```
+. <(eksctl completion bash)
+```
+
+Or for zsh:
+```
+. <(eksctl completion zsh)
```
Comment (rndstr) [inline README.md]: i'm also fairly new to zsh completion but i just tried it and i don't think that's how it's going to work unfortunately.

My brief research has led me to something like this

```
mkdir ~/.zsh/completion/
eksctl completion zsh > ~/.zsh/completion/_eksctl
```
Then add this to the `fpath` in `~/.zshrc`
```
 fpath=(~/.zsh/completion $fpath)
```

File: pkg/ctl/completion/completion.go [hunk 2]
```diff
@@ -33,10 +33,12 @@ To configure your bash shell to load completions for each session add to your ba
 
 . <(eksctl completion zsh)
 
-To configure your zsh shell to load completions for each session add to your zshrc
+To configure your zsh shell, run:
+
+mkdir -p ~/.zsh/completion/
+eksctl completion zsh > ~/.zsh/completion/_eksctl
+fpath=($fpath ~/.zsh/completion)
```
Comment (rndstr) [inline pkg/ctl/completion/completion.go]: maybe expand on this that it should be put into `~/.zshrc`?

File: pkg/ctl/completion/completion.go [hunk 3]
```diff
@@ -33,10 +33,12 @@ To configure your bash shell to load completions for each session add to your ba
 
 . <(eksctl completion zsh)
```
Comment (rndstr) [inline pkg/ctl/completion/completion.go]: this isn't correct anymore

**Comment by errordeveloper** [comment]
LGTM, thanks a lot!

**Pull request merged**